### PR TITLE
SITL: add Omni3Mecanum vehicle to SIM_Rover

### PR DESCRIPTION
**Pull request by srmainwaring**
Add an Omni3 Mecanum vehicle to SIM_Rover

Partially completes https://github.com/ArduPilot/ardupilot/issues/11782

### Details

- Add new frame type `rover-omni3mecanum` to sim_vehicle.
- Add matching param file.
- Add omni3 specific member variables to SIM_Rover
- Implement dynamics update for the omni3 rover.
- Split update method into parts that are frame specific and a part common to all frames.

Note the dynamics in SIM_Rover is not an independent test of the Omni3Mecanum mixer as it uses the same kinematics calculation for mapping body motion to wheel angular velocities.

### Testing

Run SITL with the new frame type
```bash
sim_vehicle.py --debug -v Rover -f rover-omni3mecanum --enable-dds --console --map
```

Move the rover in `MANUAL` mode using the RC override commands:

- yaw right: `rc 1 1550`, yaw left: `rc 1 1450`

omni_sitl_yaw

- move forward: `rc 3 1600`, move back: `rc 3 1400`

omni_sitl_fwd

- move right: `rc 4 1600`, move left: `rc 4 1400`

omni_sitl_right

### CODE REVIEW

**Review by IamPete1 — APPROVED** [review]
LGTM, thanks.

File: Tools/autotest/default_params/rover-omni3mecanum.parm [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ATC_STR_RAT_D 0.000010
+ATC_STR_RAT_FF 0.100000
+ATC_STR_RAT_I 0.010000
+ATC_STR_RAT_MAX 360.000000
+ATC_STR_RAT_P 0.030000
+CRUISE_SPEED 1.0
+CRUISE_THROTTLE 42.0
+FRAME_TYPE 4
+PSC_VEL_P 0.500000
+RC1_MAX 2000
```
Comment (rmackay9) [inline Tools/autotest/default_params/rover-omni3mecanum.parm:13]: Hi @srmainwaring, thanks for this.  Normally we don't add items to the param file that are the same as the defaults.  I suspect we don't need these RCx_MIN/MAX values nor the SERVOx_MIN/TRIM/MAX

Comment (srmainwaring) [reply Tools/autotest/default_params/rover-omni3mecanum.parm]: Thanks @rmackay9, updated in https://github.com/ArduPilot/ardupilot/commit/0044bb56d5b5bf2d7e206e845de4d3d00eafedd9. All remaining params are different from defaults or not set in `rover.param`.

**Review by peterbarker — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thjanks!

**Pull request merged**